lease.go

```diff
@@ -134,16 +140,16 @@ func (lm *FileLeaseManager) syncWgRecords() error {
 	return nil
 }
 
-func makePeerConfig(pubKey string, record *WgRecord) (*wgtypes.PeerConfig, error) {
+func makePeerConfig(record *WgRecord) (*wgtypes.PeerConfig, error) {
```

Comment:
Why not `func (r *WgRecord) PeerConfig() (*wgtypes.PeerConfig, error)` ?

Reply:
Not a part of this PR (I mean I am not touching the name of the func), but since I am here I could change the name of the function as well.

Reply:
What I mean is, now that your only argument is the WgRecord itself, it's a good fit for a method on the struct.

Reply:
I don't follow that logic, tbh. The difference is that the pub key is now a part of the WgRecord struct (which is still an argument). I do not see how translating a record to a wg peer config now fits better into the lease manager struct.